pkg/landscaper/controllers/execution/controller.go

```diff
@@ -75,13 +76,13 @@ func (c *controller) reconcileNew(ctx context.Context, req reconcile.Request) (r
 
 	if exec.DeletionTimestamp.IsZero() && !kubernetes.HasFinalizer(exec, lsv1alpha1.LandscaperFinalizer) {
 		controllerutil.AddFinalizer(exec, lsv1alpha1.LandscaperFinalizer)
-		if err := c.Writer().UpdateExecution(ctx, read_write_layer.W000086, exec); err != nil {
+		if err := c.Writer(ctx).UpdateExecution(ctx, read_write_layer.W000086, exec); err != nil {
```

Comment:
I'm not sure if I like this construct because it differs from the general pattern and fails if someone reuses a writer. 

```
w := c.Writer(ctx)
w.UpdateExecution()
....
w.UpdateExecution()
```

Reply:
I have reverted this, so that the `Writer` method has no argument.